CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -362,6 +362,12 @@ + (void)initWithChannelId {
     
     if (autoRegister) {
         [self subscribe];
+    } else {
+        if (![self isSubscribed]) {
```

Comment:
What is the purpose of this in the init function?

Reply:
@gbhrdt the issue we were facing it was because of the `auto Register` `false` & once autoRegister's value gets `false` it will break the flow and not go for the `[self subscribe];` and all the core logic of the topics had been written inside the success block of the notification permission and based on the selected topics we have written entire `CPTopicsViewController` such as `BOOL selectedState = (([selectedTopics count] == 0 && !hasTopics && !defaultUnchecked) || [selectedTopics containsObject:[topic id]]);` and many other functions in that file so to pre-fill the selected topics I have put the check and write down that logic.

Reply:
But inside `initialisedTopicDialogData` you are calling `setSubscriptionTopics` which is then triggering a subscribe. This is not what we want, because then `autoRegister: false` is useless and will always trigger a subscribe?
We only need to execute this logic inside `showTopicsDialog` not right in the init. And also the logic should not call `setSubscriptionTopics` to prevent a subscribe.